compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/EnvironmentResolver.java

```diff
@@ -605,6 +610,19 @@ public void visit(BLangOnFailClause onFailClause) {
         this.acceptNode(onFailClause.body, this.symbolEnv);
     }
 
+    @Override
+    public void visit(BLangArrowFunction arrowFunctionNode) {
+        SymbolEnv arrowFunctionSymbolEnv = SymbolEnv.createArrowFunctionSymbolEnv(arrowFunctionNode, this.symbolEnv);
+        arrowFunctionNode.params.forEach(param -> symbolEnter.defineNode(param, arrowFunctionSymbolEnv));
```

Comment:
We shouldn't be using nor do we need to use `SymbolEnter` here. This means you are modifying the AST and potentially altering semantics.